packages/yarnpkg-core/sources/StreamReport.ts

```diff
@@ -496,34 +509,37 @@ export class StreamReport extends Report {
 
     const spinner = PROGRESS_FRAMES[this.progressFrame];
 
-    const styleName = this.configuration.get(`progressBarStyle`) || defaultStyle;
-    if (!Object.prototype.hasOwnProperty.call(PROGRESS_STYLES, styleName))
-      throw new Error(`Assertion failed: Invalid progress bar style`);
-
-    const style = PROGRESS_STYLES[styleName];
-
-    const PAD_LEFT = `➤ YN0000: ┌ `.length;
-
-    const maxWidth = Math.max(0, Math.min(process.stdout.columns - PAD_LEFT, 80));
-    const scaledSize = Math.floor(style.size * maxWidth / 80);
-
     for (const {progress} of this.progress.values()) {
-      const okSize = scaledSize * progress;
+      const okSize = this.progressMaxScaledSize * progress;
 
-      const ok = style.chars[0].repeat(okSize);
-      const ko = style.chars[1].repeat(scaledSize - okSize);
+      const ok = this.progressStyle.chars[0].repeat(okSize);
+      const ko = this.progressStyle.chars[1].repeat(this.progressMaxScaledSize - okSize);
 
       this.stdout.write(`${formatUtils.pretty(this.configuration, `➤`, `blueBright`)} ${this.formatName(null)}: ${spinner} ${ok}${ko}\n`);
     }
 
     this.progressTimeout = setTimeout(() => {
+      this.progressLastScaledSize = -1;
       this.refreshProgress();
-    }, 1000 / 60);
+    }, 1000 / 10);
   }
 
   private refreshProgress(delta: number = 0) {
-    this.clearProgress({delta});
-    this.writeProgress();
+    if (this.progress.size === 0) {
+      this.clearProgress({delta});
+      this.writeProgress();
+    }
+
+    for (const {progress} of this.progress.values()) {
+      const progressScaledSize = Math.trunc(this.progressMaxScaledSize * progress);
+
+      if (progressScaledSize !== this.progressLastScaledSize) {
+        this.clearProgress({delta});
+        this.writeProgress();
+
+        this.progressLastScaledSize = progressScaledSize;
+      }
```

Comment:
I don't think that works - since there can be multiple progressbars active at the same time (like in the doctor), you can't store the scaled size in the `this.progressLastScaledSize`, otherwise only the last one will be accounted for, it could have weird artifacts 🤔

Reply:
I think you should instead tweak `ProgressDefinition` to contain the last rendered scaled progress, and compare that independently for each progress bar. And set a boolean during the iteration (instead of immediately printing) so that `writeProgress` isn't called multiple times during the same `refreshProgress` call.

Reply:
Thank you for the hints @arcanis ! Done